code/game/objects/items/weapons/barricade_kit.dm

```diff
@@ -97,6 +97,9 @@
 			if(istype(S, /obj/structure/window/barricade/full/block))
 				to_chat(user, "<span class='warning'>There already is a barricade here</span>")
 				return
+			if(istype(A, /obj/structure/window/barricade))
```

Comment:
Yes, it would work
That's why I said "[not] quite the right way" rather than "the wrong way"